workspaces/src/rpc/query.rs

```diff
@@ -0,0 +1,393 @@
+//! This module defines a bunch of internal types used solely for querying into RPC
+//! methods to retrieve info about what's on the chain. Note that the types defined
+//! exposed as-is for users to reference in their own functions or structs as needed.
+//! These types cannot be created outside of workspaces. To use them, refer to
+//! surface level types like [`Account`], [`Contract`] and [`Worker`].
+//!
+//! For example, to query into downloading
+//! contract state:
+//! ```ignore
```

Comment:
would be nice if this wasn't ignored so that the interface is run through doc tests and this doesn't get outdated if changes, nbd though